src/vm/virtualMachine.ts

```diff
@@ -0,0 +1,46 @@
+import { MemorySegmentManager } from 'memory/memory';
+import { Felt } from 'primitives/felt';
+import { Uint, UnsignedInteger } from 'primitives/uint';
+import { Result, Err, VMError } from 'result-pattern/result';
+import { RunContext } from 'run-context/runContext';
+
+export const InstructionError = {
+  message: 'VMError: VM Instruction must be a Field Element',
+};
+
+export class VirtualMachine {
+  private runContext: RunContext;
+  private currentStep: Uint;
+  private segments: MemorySegmentManager;
+
+  constructor() {
+    this.currentStep = UnsignedInteger.toUint(0n);
+    this.segments = new MemorySegmentManager();
+    this.runContext = new RunContext();
+  }
+
+  step(): Result<true, VMError> {
+    const maybeEncodedInstruction = this.segments.memory.get(
+      this.runContext.getPc()
+    );
+
+    if (maybeEncodedInstruction.isErr()) {
+      return maybeEncodedInstruction;
+    }
+
+    const encodedInstruction = maybeEncodedInstruction.unwrap();
+
+    if (!(encodedInstruction instanceof Felt)) {
+      return new Err(InstructionError);
+    }
+
+    const maybeUint = encodedInstruction.toUint();
```

Comment:
I don't think we need to call toUint, I think the instruction needs to be limited to Uint64

Reply:
problem is that JS's max integer is 2^53 so that's like U53? 
We need bigint for those 11 last bits,

Reply:
I could create a U64 type though like Uint, but with a higher bound

Reply:
Would some helpers on Uint work? We could just have something like `isUint64`?

Reply:
Mmhm maybe but we'd need a new type for typeguards to truly be useful